new-components/targets/git-clone/pkg/git/git.go

```diff
@@ -0,0 +1,207 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/go-errors/errors"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/smithy-security/pkg/env"
+)
+
+const errInvalidConfigurationReasonEmpty = "cannot be empty"
+
+type (
+	// Conf wraps the component configuration.
+	Conf struct {
```

Comment:
this is also missing a set of dirs for git sparse checkout, useful for monorepos

Reply:
Going to skip this for now as discussed in person